scrapy/spiders/__init__.py

```diff
@@ -57,24 +54,8 @@ def _set_crawler(self, crawler):
         crawler.signals.connect(self.close, signals.spider_closed)
 
     def start_requests(self):
-        cls = self.__class__
-        if method_is_overridden(cls, Spider, 'make_requests_from_url'):
-            warnings.warn(
-                "Spider.make_requests_from_url method is deprecated; it "
-                "won't be called in future Scrapy releases. Please "
-                "override Spider.start_requests method instead (see %s.%s)." % (
-                    cls.__module__, cls.__name__
-                ),
-            )
-            for url in self.start_urls:
-                yield self.make_requests_from_url(url)
-        else:
-            for url in self.start_urls:
-                yield Request(url, dont_filter=True)
-
-    def make_requests_from_url(self, url):
-        """ This method is deprecated. """
-        return Request(url, dont_filter=True)
+        for url in self.start_urls:
+            yield Request(url, dont_filter=True)
```

Comment:
There is a problem in a way we deprecated `make_requests_from_url` method: if user calls this method to construct a Request, no deprecation warning is raised. This may happen in overridden start_requests, or in any other callback. 

So it looks like it was never properly deprecated :( A common case was handled, but people were using it in a different way as well (as shown by our internal codesearch tool). 

I'm not sure what to do, but have a slight preference for starting to raise warnings in the `make_requests_from_url` method, instead of removing it. If we do this, it is probably better to keep compatibility code in start_requests as well: it seems that removing its usage in start_requests without removing the method itself can lead to unnoticed bugs in user code.

Reply:
I see your point, but I wonder if that isn't the right thing to do. I believe this is similar to https://github.com/scrapy/scrapy/pull/4170#issuecomment-556033732, in the sense that we shouldn't prevent users from defining any method they want in their spiders. Mi interpretation is that the deprecation meant the method would cease to be used from the _default_ `start_requests` implementation. Thoughts?